spec/mem/tree_borrows/memory.md

```diff
@@ -109,16 +107,32 @@ impl<T: Target> TreeBorrowsMemory<T> {
             ret::<Result<Path>>(child_path)
         })?;
 
+        // Track the new protector
+        if protected.yes() { self.add_protector(call_id, (alloc_id, child_path)); }
+
         // Create the child pointer and return it 
         ret(Pointer {
             provenance: Some((alloc_id, child_path)),
             ..ptr
         })
     }
 
+    /// Remove the protector.
+    /// `provenance` is the provenance of the protector.
+    /// Perform a special implicit access on all locations that have been accessed.
+    fn release_protector(&mut self, provenance: TreeBorrowsProvenance) -> Result {
+        let (alloc_id, path) = provenance;
+        self.allocations.mutate_at(alloc_id.0, |allocation| {
+            // Special case for the `Box`, since the `Box` could be deallocated during the execution of the function
```

Comment:
```suggestion
            // If this is weakly protected, it could have been already deallocated.
            // In that case, we don't do anything here -- fake reads from freed memory don't
            // make a lot of sense.
```